src/commands/StepCommand.java

```diff
@@ -32,6 +34,11 @@ public void execute(String[] input, SimulationManager simulationManager) {
             return;
         }
 
+        if (simulationManager.atTraceEnd()) {
```

Comment:
I don't think adding a new public method for such a specific use case is the best thing to do.

I think it would be better to have this check within `SimulationManager.performStep`.

Reply:
Good call.